webapp/src/components/post_type/action_view/action_view.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ActionButton from './action_button';
+
+export default class ActionView extends React.PureComponent {
+    static propTypes = {
+        post: PropTypes.object.isRequired,
+        attachment: PropTypes.object.isRequired,
+        votedAnswers: PropTypes.object.isRequired,
+        siteUrl: PropTypes.string.isRequired,
+
+        actions: PropTypes.shape({
+            fetchVotedAnswers: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    componentDidMount() {
+        this.props.actions.fetchVotedAnswers(this.props.siteUrl, this.props.post.props.poll_id);
+    }
+
+    render() {
+        const actions = this.props.attachment.actions;
+        if (!actions || !actions.length) {
+            return '';
+        }
+
+        const content = [];
+        const votedAnswers = this.props.votedAnswers || {};
+        const answers = votedAnswers[this.props.post.props.poll_id] || {};
+
+        actions.forEach((action) => {
+            if (!action.id || !action.name) {
+                return;
+            }
+            switch (action.type) {
+            case 'button':
+                content.push(
+                    <ActionButton
+                        key={action.id}
+                        action={action}
+                        postId={this.props.post.id}
+                        hasVoted={answers.voted_answers && answers.voted_answers.includes(action.name)}
```

Comment:
IE doesn't not support `includes`, if this is desired to be supported `indexOf` can be used instead.

Reply:
https://caniuse.com/#search=includes

Reply:
I didn't know that, thanks!